lib/core/nav/HeaderNav.js

```diff
@@ -202,20 +205,16 @@ class HeaderNav extends React.Component {
       // set link to page with current page's language if appropriate
       const language = this.props.language || '';
       if (fs.existsSync(CWD + '/pages/en/' + link.page + '.js')) {
-        href =
-          siteConfig.baseUrl +
-          (language ? language + '/' : '') +
-          link.page +
-          extension;
+        href = urljoin(siteConfig.baseUrl, language, link.page, extension);
```

Comment:
This can cause a bug
1. If `link.page` is `doc` and `extension` is `.html`, it will become `doc/.html` instead of `doc.html`
2. url-join expect string, if language is undefined it will throw an error. 
Try running
```js
console.log(urljoin('http://www.google.com', undefined, '/b/cd', '?foo=123'));
```


Use this instead
```js
urljoin(siteConfig.baseUrl, language || '', link.page + extension);
```